src/crypto/secretbox/error.rs

```diff
@@ -0,0 +1,25 @@
+//! Error types for the `secretbox` module
+
+/// A type representing error conditions of secret box operations
+#[derive(Debug, PartialEq)]
+pub enum SecretBoxError {
+    /// The ciphertext was too short. This results from using the `crypto_secretbox_open_easy`
+    /// function, which expects a message prepended with a MAC.
+    CiphertextTooShort,
+    /// The ciphertext failed verification.
+    CiphertextFailedVerification,
+}
+
+#[cfg(feature = "std")]
+impl std::fmt::Display for SecretBoxError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for SecretBoxError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
```

Comment:
It's a default implementation, isn't it? https://doc.rust-lang.org/src/std/error.rs.html#100 So the whole block is equivalent to `impl std::error::Error for SecretBoxError {}`